node/pkg/chain/websocketchainreader/type.go

```diff
@@ -15,6 +15,32 @@ const (
 	Ethereum BlockchainType = 2
 )
 
+type ChainReaderConfig struct {
+	KaiaWebsocketUrl string
+	EthWebsocketUrl  string
+	RetryInterval    time.Duration
+}
+
+type ChainReaderOption func(*ChainReaderConfig)
+
+func WithKaiaWebsocketUrl(url string) ChainReaderOption {
+	return func(c *ChainReaderConfig) {
+		c.KaiaWebsocketUrl = url
+	}
+}
+
+func WithEthWebsocketUrl(url string) ChainReaderOption {
+	return func(c *ChainReaderConfig) {
+		c.EthWebsocketUrl = url
+	}
+}
+
+func WithRetryInterval(interval time.Duration) ChainReaderOption {
+	return func(c *ChainReaderConfig) {
+		c.RetryInterval = interval
+	}
+}
+
```

Comment:
how about moving functions to somewhere else, like `utils` file?

Reply:
It's function related to struct initialization so I included in `type.go`. If you insist, I'll move it into utils.go within the same folder though.  

And thought it could be kept it as it is since it is still short  🙏

Reply:
ah i see then lets keep it as it is